datasource/local/src/iosMain/kotlin/ke/don/local/datastore/IosProfileStore.kt

```
@@ -0,0 +1,39 @@
+    private val profileKey = PROFILE_KEY
+    private val json = Json { ignoreUnknownKeys = true }
+
+    private val _profileFlow = MutableStateFlow(loadProfile())
+    override val profileFlow: Flow<Profile?> = _profileFlow.asStateFlow()
+
+    private fun loadProfile(): Profile? {
+        val stored = defaults.stringForKey(profileKey) ?: return null
+
+        return try {
+            json.decodeFromString<Profile>(stored)
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)